### PR TITLE
INT-3759: Ability to Disable Logging on Main Path

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INT-3759

Add `IntegrationManagementConfigurer` to globally set the default.

This will be documented/expanded when JIRAs 3755/3756 are implemented.

### CODE REVIEW

File: spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagement.java [hunk 1]
```diff
@@ -19,13 +19,19 @@
...
+ * Base interface for Integration managed components.
  *
  * @author Gary Russell
  * @since 4.2
  *
  */
-public interface CountsEnablement {
+public interface IntegrationManagement {
+
+	@ManagedAttribute(description = "Use to disable all debug logging on the happy path")
```
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagement.java]: Looks like not technical sentence, but I can live with that if you insist that it isn't important and we go ahead with other work :smile:

File: spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	private boolean defaultLoggingEnabled = true;
+
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		this.applicationContext = applicationContext;
+	}
+
+	/**
+	 * Disable all logging in the main message flow in framework components. When 'false', such logging will be
```
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java]: Sorry, what is "non-main" then?

File: spring-integration-jmx/src/main/java/org/springframework/integration/monitor/LifecycleMessageHandlerMetrics.java [hunk 3]
```diff
@@ -164,6 +164,16 @@ public boolean isCountsEnabled() {
 	}
 
 	@Override
+	public void setLoggingEnabled(boolean enabled) {
+		delegate.setLoggingEnabled(enabled);
```
Comment (artembilan) [inline spring-integration-jmx/src/main/java/org/springframework/integration/monitor/LifecycleMessageHandlerMetrics.java]: Minor: should be with `this.`, but anyway...

File: spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * Configures beans that implement {@link IntegrationManagement}.
+ *
+ * TODO: This class will be expanded by INT-3755/3756.
+ *
+ * @author Gary Russell
+ * @since 4.2
+ *
+ */
+public class IntegrationManagementConfigurer implements SmartInitializingSingleton, ApplicationContextAware {
```
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java]: I don't see how this component is used from Framework.
I think we can expose all options from here to the `integration.properties` and populate a bean for this automatically.

**Comment by artembilan** [comment]
Otherwise it's OK. 
We can merge as is and go ahead with other concerns afterwards

File: spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java (hunk 4 above)
Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java]: > * TODO: This class will be expanded by INT-3755/3756.

INT-3755 moves the stats configuration from the IMBE to this class so stats can be captured independently of JMX.

Please wait for my next PR (post M2) - if you want a preview, see https://github.com/garyrussell/spring-integration/commit/39729ea79c12e596c8d2164ee7af9dcf7202e0a3#diff-b8dc44d60dfa08db849a57eb3b07134b

Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java]: ???
Do we merge this PR or not for the M2?
Should I take a look to the fix or go ahead with Zookeeper PRs ?

Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java]: Yes; this needs to be merged, I was just trying to explain that while this class doesn't look very useful now, it is the basis for extracting the management from the IMBE which will be in my next PR (after M2).

File: spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+package org.springframework.integration.support.management;
+
+import java.util.Map;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.SmartInitializingSingleton;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import reactor.core.support.Assert;
```
Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java]: Bad import. Will be fixed on merge

File: spring-integration-jmx/src/test/java/org/springframework/integration/jmx/MBeanAttributeFilterTests.java [hunk 6]
```diff
@@ -105,7 +105,7 @@ public void testAttributeFilterNot() {
 		Map<String, Object> bean = (Map<String, Object>) payload
 				.get(domain + ":name=in,type=MessageChannel");
 
-		assertEquals(8, bean.size());
+		assertEquals(9, bean.size());
```
Comment (artembilan) [inline spring-integration-jmx/src/test/java/org/springframework/integration/jmx/MBeanAttributeFilterTests.java]: It isn't obvious for normal people, so let's add all those 9 attributes to test!

Comment (garyrussell) [inline spring-integration-jmx/src/test/java/org/springframework/integration/jmx/MBeanAttributeFilterTests.java]: good point; I'll push a fix

**Comment by garyrussell** [comment]
Rebased, pushed

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/9cc0652b611ca3079646cf9f8c509b69b7781cb1

**Pull request closed without merging**